### PR TITLE
Fix/imap connector delete message

### PR DESCRIPTION
**Pull request by clemensgrosse**
# Description

- fix of delete message in AbstractInboxConnector
- deletion of deprecated methods in AbstractInboxConnector
- adding usage of latest testerra code for MailConnectorTest (current methods of AbstractInboxConnector, PropertyManagerProvider, ASSERT)
- correcting T11 in MailConnectorTest

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: mail-connector/src/main/java/eu/tsystems/mms/tic/testframework/mailconnector/util/AbstractInboxConnector.java [hunk 1]
```diff
@@ -609,7 +530,8 @@ private boolean pDeleteMessage(final SearchTerm searchTerm, final String folderN
                 messages = folder.getMessages();
             }
 
-            log().info("Checking messages from " + getInboxFolder() + " for MessageID:");
+            final String logMessageSearchTerm = String.format("Checking messages from '%s' for MessageID: '%s'", getInboxFolder(), searchTerm);
+            log().info(logMessageSearchTerm);
```
Comment (martingrossmann) [inline mail-connector/src/main/java/eu/tsystems/mms/tic/testframework/mailconnector/util/AbstractInboxConnector.java:534]: Please use default logging syntax with `{}`, e.g. `log().info("Checking messages from {}", getInboxFolder())`

**Comment by martingrossmann** [comment]
Please recheck the mail connector docs if there is needed an update.

File: mail-connector/src/test/java/eu/tsystems/mms/tic/testframework/mailconnector/test/MailConnectorTest.java [hunk 2]
```diff
@@ -703,7 +718,14 @@ private Email waitForMessage(final SearchTerm searchTerm, final AbstractInboxCon
 
         // TEST - Fail, if no message was received.
         try {
-            receivedMsg = abstractInboxConnector.waitForMails(searchTerm, folderName).get(0);
+            final EmailQuery query = new EmailQuery()
+                    .setSearchTerm(searchTerm)
+                    .setFolderName(folderName);
+            final List<Email> emailList = abstractInboxConnector.query(query).collect(Collectors.toList());
+            if (emailList.isEmpty()) {
```
Comment (martingrossmann) [inline mail-connector/src/test/java/eu/tsystems/mms/tic/testframework/mailconnector/test/MailConnectorTest.java:725]: Please use an `Assert` to check for `emailList.isEmpty`. I think the `try...catch` is not needed here.

**Review by martingrossmann — CHANGES_REQUESTED** [review]

File: mail-connector/src/main/java/eu/tsystems/mms/tic/testframework/mailconnector/util/EmailQuery.java [hunk 3]
```diff
@@ -42,7 +42,7 @@ public class EmailQuery implements PropertyManagerProvider {
     private static final String MAX_READ_TRIES_PROPERTY = "MAX_READ_TRIES";
 
     private int retryCount = Integer.parseInt(PROPERTY_MANAGER.getProperty(MAX_READ_TRIES_PROPERTY, "20"));
-    private long pauseMs = Integer.parseInt(PROPERTY_MANAGER.getProperty(POLLING_TIMER_SECONDS_PROPERTY, "10")) * 1000L;
+    private long pauseMs = PROPERTY_MANAGER.getLongProperty(POLLING_TIMER_SECONDS_PROPERTY, "10") * 1000;
```
Comment (martingrossmann) [inline mail-connector/src/main/java/eu/tsystems/mms/tic/testframework/mailconnector/util/EmailQuery.java:45]: Default value should not `"10"`, but `10_000L` and the calculation with `* 1000` ist not needed any more.

**Review by mbeuthan — APPROVED** [review]

**Review by martingrossmann — APPROVED** [review]

**Pull request merged**